### PR TITLE
minor bugfixes

### PR DESCRIPTION
**Pull request by weirichs**
existsBackgroundVariables: if variable column is provided as numeric, it must bei integer. halveString: patterns with more than one character are allowed

### CODE REVIEW

File: R/existsBackgroundVariables.r [hunk 1]
```diff
@@ -15,6 +15,7 @@ existsBackgroundVariables <- function(dat, variable, warnIfMissing = FALSE )  {
                   			varColumn <- match(variable, colnames(dat))
                  	   }
                      if(is.numeric(variable))   {
+                        if ( !all(round(variable) == variable)) {stop(paste0("If provided as column number, variable '",variable,"' must be integer."))}
```
Comment (beckerbenj) [inline R/existsBackgroundVariables.r:18]: `paste0()` could simply be omitted here (`stop()` pastes automatically).

File: R/remove_funs.R [hunk 2]
```diff
@@ -32,12 +32,12 @@ halveString <- function (string, pattern, first = TRUE, colnames=c("X1", "X2") )
   if(first)  {
     ret <- as.matrix(data.frame(X1 = unlist(lapply(allSplit, FUN = function (l) { l[1]})),
                                 X2 = unlist(lapply(1:length(allSplit), FUN = function (l) {
-                                  if(length(allSplit[[l]]) == 1 )  { ret <- NA } else { ret <- substring(string[l], first = nchar(allSplit[[l]][1])+2 ) }
+                                  if(length(allSplit[[l]]) == 1 )  { ret <- NA } else {ret <- substring(string[l], first = nchar(allSplit[[l]][1])+1+nchar(pattern) ) }
```
Comment (beckerbenj) [inline R/remove_funs.R:35]: not sure what happens here exactly; but seems reasonable?

File: man/halveString.rd [hunk 3]
```diff
@@ -28,7 +28,7 @@ Optional: character vector of length 2 to specify the colnames of the resulting
 }
 }
 \value{
-A data.frame with two columns
+A matrix with two columns
```
Comment (beckerbenj) [inline man/halveString.rd:31]: lgtm

**Review by beckerbenj — CHANGES_REQUESTED** [review]
Sorry if those are nitpicks!

File: NEWS.md [hunk 4]
```diff
@@ -1,3 +1,5 @@
+* adapt `halveString()` for patterns with more than 1 character
```
[the remark this answers is not in the document]
Comment (beckerbenj) [reply NEWS.md:1]: also increase version number in description to 0.7.4.9000

File: R/remove_funs.R (hunk 2 above)
Comment (beckerbenj) [reply R/remove_funs.R:35]: tests should be added for `halveString()` in general, but I can do this later

File: R/existsBackgroundVariables.r (hunk 1 above)
Comment (weirichs) [reply R/existsBackgroundVariables.r:18]: will be done

File: R/remove_funs.R (hunk 2 above)
Comment (weirichs) [reply R/remove_funs.R:35]: If the string "a__b__c" should be splitted at first occurrence of pattern "__", the result should be c("a", "b__c"). First element of result equals first element of strsplit("a__b__c"). For the second element, however, we must use substring("a__b__c", first = x). And x equals the sum of nchar(unlist(strsplit("a__b__c", "__"))[1]) + nchar("__").

File: NEWS.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (weirichs) [reply NEWS.md:1]: will be done

**Pull request merged**